internal/dag/dag.go

```diff
@@ -672,6 +672,9 @@ type PeerValidationContext struct {
 	// SubjectName holds an optional subject name which Envoy will check against the
 	// certificate presented by the upstream.
 	SubjectName string
+	// SubjectNames holds optional subject names which Envoy will check against the
+	// certificate presented by the upstream. The first entry must match the value of SubjectName
+	SubjectNames []string
```

Comment:
hmmm interesting. I'm on board to try it out. However, one question.

What do you think about this in relation to extension services?

I see there is this line: https://github.com/projectcontour/contour/blob/main/internal/dag/extension_processor.go#L177

Does the `SNI` field on the `ExtensionCluster` struct need to be singular, or should that be switched to allow multiple?

Reply:
or is it safe to default to the first entry in SubjectNames?

Reply:
I think for the purposes of this PR it can stay singular and you can just assign the first subject name from UpstreamValidation to it. The issue referenced in the comment (https://github.com/projectcontour/contour/issues/2893) can track if we need some other mechanism to set the SNI  for ExtensionServices explicitly.